transforms/no-vars.js

```diff
@@ -38,7 +38,15 @@ export default function(file, api) {
       );
     } else if (id.type === 'ArrayPattern') {
       return id.elements.map(
-        e => (e.type === 'RestElement' ? e.argument.name : e.name)
+        e => {
+          if (!e) {
+            return null;
+          }
+          if (e.type === 'RestElement') {
+            return e.argument.name;
+          }
+          return e.name;
+        }
       );
```

Comment:
You could add a `.filter(name => name !== null)` here to avoid the `null` checks elsewhere.